### DISCUSSION TITLE
How to implement built-in service in the router?

### DISCUSSION THREAD

**Discussion by giautm** [apollographql/router#1167] [General]
Hi, I want to migrate our project from using apollo-federation 2 to a router. But, I have a service that is built-in in the GW. It's Relay Node service. You can see the implementation in this project https://github.com/victorandree/apollo-federation-relay/blob/master/node-gateway.js

I need to do something similar in apollo/router. But I don't know where I can start to integrate it. Any guide/comment will be helpful. Thank you.

**Comment by deweyjose** [comment]
We also have this use-case internally.

**Comment by gtkatakura** [comment]
Same here. It is the currently block to move on to Apollo Router
